### PR TITLE
Validation branch

### PR DESCRIPTION
**Pull request by leonardoavs**
This is a branch that has the validation logic for the String type, has all the functionality for the database files, the html files and so on. 

Let me know if you like or if you want changes on the code. I can apply the changes for all types and make them more general. 

Thanks.

### CODE REVIEW

**Comment by jdubois** [comment]
- yes you need to rebase, you have some old code here
- and as we said in the original ticket, can you add a first question to ask the user if he wants validation? Because this adds quite a lot of new questions, and not everybody might want to answer all those
- this is clearly my next iteration -> can you work on this in the next couple of days? Otherwise I'll start from your work

File: entity/templates/src/main/resources/config/cql/_added_entity.cql [hunk 1]
```diff
@@ -3,7 +3,7 @@ CREATE TABLE IF NOT EXISTS <%= entityInstance %> (
     id uuid,<% for (fieldId in fields) { %><% if (fields[fieldId].fieldType == 'UUID') { %>
     <%= fields[fieldId].fieldName %> uuid,<% } else if (fields[fieldId].fieldType == 'TimeUUID') { %>
     <%= fields[fieldId].fieldName %> timeuuid,<% } else if (fields[fieldId].fieldType == 'String') { %>
-    <%= fields[fieldId].fieldName %> text,<% } else if (fields[fieldId].fieldType == 'Integer') { %>
+    <%= fields[fieldId].fieldName %> varchar(<%=fields[fieldId].fieldLength%>) <% if (fields[fieldId].fieldRequired) { %>NOT NULL<% } %>,<% } else if (fields[fieldId].fieldType == 'Integer') { %>
```
Comment (jdubois) [inline entity/templates/src/main/resources/config/cql/_added_entity.cql]: I don't think this is correct with Cassandra, did you test this?

**Comment by jdubois** [comment]
Hi, I'm sorry but I might not take your PR: don't take it personally, I had an electrical failure at home, so no Internet, and while I was staying in the dark I coded another version (I was remembering some part of what you did, but without Internet I couldn't merge anything, obviously!!)
Now that everything is back to normal I'm going to have a closer at what you did, of course, and cherry pick stuff if possible

**Comment by jdubois** [comment]
So I've messed up your PR with my new commit -> at the moment I'm just asking the validation questions, see https://github.com/jhipster/generator-jhipster/commit/19dfb0137eeff08ee78fa732ffe0eda7f0a28650
But the rest of the code is ready to be transformed into templates

**Comment by leonardoavs** [comment]
@jdubois, @gzsombor  Ok, no problem the intention was to contribute my code, is good to see that the validations have been improved, and in a future version of the project we will have validations.

Sorry for not response before, but I was too busy. 

Thanks.

File: entity/templates/src/main/resources/config/cql/_added_entity.cql (hunk 1 above)
Comment (leonardoavs) [inline entity/templates/src/main/resources/config/cql/_added_entity.cql]: Yes this is not correct, my bad, I forgot that cassandra does not support fixed length varchars.

**Pull request closed without merging**